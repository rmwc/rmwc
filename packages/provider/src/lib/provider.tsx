import React from 'react';
import * as RMWC from '@rmwc/types';

type TooltipActivationT = 'hover' | 'click' | 'focus';

type RCTooltipAlignT =
  | 'left'
  | 'right'
  | 'top'
  | 'bottom'
  | 'topLeft'
  | 'topRight'
  | 'bottomLeft'
  | 'bottomRight';

type TooltipAlignT =
  | 'start'
  | 'center'
  | 'end'
  | 'above'
  | 'below'
  | 'startAbove'
  | 'startBelow'
  | 'centerAbove'
  | 'centerBelow'
  | 'endAbove'
  | 'endBelow';

export type TooltipOptions = {
  /** How to align the tooltip. This affects both RCTooltip and Tooltip, but only if the given alignment is supported by the component. Defaults to `top`. */
  align?: RCTooltipAlignT | TooltipAlignT;
  /** Activate the tooltip through one or more interactions. Defaults to `['hover', 'focus']`. */
  activateOn?: TooltipActivationT | TooltipActivationT[];
  /** Whether or not to show an arrow on the Tooltip. Only supported by RCTooltip. Defaults to `false`. */
  showArrow?: boolean;
  /** Delay in milliseconds before showing the tooltip when interacting via touch or mouse. */
  enterDelay?: number;
  /** Delay in milliseconds before hiding the tooltip when interacting via touch or mouse. */
  leaveDelay?: number;
};

type TypographyOptions = {
  defaultTag?: string | React.ComponentType<any>;
  headline1?: string | React.ComponentType<any>;
  headline2?: string | React.ComponentType<any>;
  headline3?: string | React.ComponentType<any>;
  headline4?: string | React.ComponentType<any>;
  headline5?: string | React.ComponentType<any>;
  headline6?: string | React.ComponentType<any>;
  subtitle1?: string | React.ComponentType<any>;
  subtitle2?: string | React.ComponentType<any>;
  body1?: string | React.ComponentType<any>;
  body2?: string | React.ComponentType<any>;
  caption?: string | React.ComponentType<any>;
  button?: string | React.ComponentType<any>;
  overline?: string | React.ComponentType<any>;
};

/** A provider for setting global options in RMWC. */
export interface RMWCProviderProps {
  /** Enable / Disable interaction ripples globally */
  ripple?: boolean;
  /** Global options for icons */
  icon?: Partial<RMWC.IconOptions>;
  /** Global tooltip options */
  tooltip?: Partial<TooltipOptions>;
  /** Global typography options */
  typography?: Partial<TypographyOptions>;
  /** Children to render */
  children?: React.ReactNode;
}

// Default provider options
const providerDefaults: RMWCProviderProps = {
  ripple: true,
  tooltip: {
    align: 'top',
    showArrow: false,
    activateOn: ['hover', 'focus'],
    enterDelay: 0,
    leaveDelay: 0
  },
  typography: {},
  icon: {
    icon: '',
    basename: 'material-icons',
    prefix: '',
    strategy: 'auto',
    render: undefined
  }
};

export const ProviderContext = React.createContext(providerDefaults);

export const useProviderContext = () => React.useContext(ProviderContext);

/** A provider for setting global options in RMWC. */
export const RMWCProvider = ({ children, ...rest }: RMWCProviderProps) => {
  return (
    <ProviderContext.Provider
      value={{
        ...providerDefaults,
        ...rest
      }}
    >
      {children}
    </ProviderContext.Provider>
  );
};
