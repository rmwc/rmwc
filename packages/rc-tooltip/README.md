# Tooltips `RMWC ADDON`

Tooltips display informative text when users hover over, focus on, or tap an element.

- Module **@rmwc/tooltip**
- Import styles:
  - Using CSS Loader
    - import '@rmwc/tooltip/styles';
  - Or include stylesheets
    - **'@rmwc/tooltip/tooltip.css'**


## Basic Usage

Wrap any component in a `Tooltip` and provide the content attribute. The only requirement is that is has a single React child, and that the child can accept `onMouseEnter`, `onMouseLeave`, `onFocus`, and `onClick` props.

```jsx
<>
  <Tooltip content="Cookies">
    <IconButton icon="star_border" />
  </Tooltip>

  <Tooltip content="Pizza">
    <IconButton icon="favorite_border" />
  </Tooltip>

  <Tooltip content="Icecream">
    <IconButton icon="mood" />
  </Tooltip>
</>
```

## Variants

```jsx
<Tooltip content="Cake" showArrow>
  <IconButton icon="cake" />
</Tooltip>
```

```jsx
<Tooltip content="Hello" align="right" open={true}>
  <IconButton icon="mood" />
</Tooltip>
```

```jsx
<Tooltip
  content={
    <div style={{ display: 'flex' }}>
      <Avatar
        src="images/avatars/captainamerica.png"
        size="large"
        name="Steve Rogers"
      />
      <div style={{ marginLeft: '0.5rem' }}>
        <b>Captain America</b>
        <div>Steve Rogers</div>
      </div>
    </div>
  }
>
  <span role="button">S. Rogers</span>
</Tooltip>
```

```jsx
<Tooltip
  /** You make something like a popover window by just styling your inner content. */
  content={
    <div
      style={{
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'center',
        background: 'white',
        width: '20rem',
        height: '8rem',
        color: 'black',
        borderRadius: '3px',
        margin: '0 -3px'
      }}
    >
      Hello World!
    </div>
  }
>
  <span role="button">Popover Window</span>
</Tooltip>
```

```jsx
<>
  <Tooltip content="Cookies" enterDelay={1000}>
    <Button label="Enter Delay" />
  </Tooltip>

  <Tooltip content="Pizza" leaveDelay={1000}>
    <Button label="Leave Delay" />
  </Tooltip>

  <Tooltip content="Icecream" enterDelay={1000} leaveDelay={1000}>
    <Button label="Both" />
  </Tooltip>
</>
```

```jsx
function AlignmentExample() {
  return [
    'left',
    'right',
    'top',
    'bottom',
    'topLeft',
    'topRight',
    'bottomLeft',
    'bottomRight'
  ].map((align) => (
    <Tooltip key={align} content={`Align: ${align}`} align={align}>
      <IconButton icon="trip_origin" />
    </Tooltip>
  ));
}
```

## Activation

By default, tooltips will activate on hover and focus. You can change this behavior by passing one or more options to the `activateOn` prop.

```jsx
<>
  <Tooltip content="Cookies" activateOn="hover">
    <Button label="Hover" />
  </Tooltip>
  ''
  <Tooltip content="Pizza" activateOn="click">
    <Button label="Click" />
  </Tooltip>
  <Tooltip content="Icecream" activateOn="focus">
    <Button label="Focus" />
  </Tooltip>
  <Tooltip content="Cake" activateOn={['hover', 'focus']}>
    <Button label="Multiple" />
  </Tooltip>
</>
```

## Usage with RMWCProvider

The RMWCProvider allows you to specify global defaults for your tooltips.

```jsx
<RMWCProvider
  tooltip={{
    align: 'right',
    activateOn: 'hover',
    showArrow: true,
    leaveDelay: 500,
    enterDelay: 0
  }}
>
  <Tooltip content="Hello World!">
    <Button label="With Provider" />
  </Tooltip>
</RMWCProvider>
```

## Tooltip
A Tooltip component for displaying informative popover information.

### Props

| Name | Type | Description |
|------|------|-------------|
| `activateOn` | `TooltipActivationT \| TooltipActivationT[]` | Activate the tooltip through one or more interactions. Defaults to `['hover', 'focus']`. |
| `align` | `TooltipAlignT` | How to align the tooltip. Defaults to `top`. |
| `children` | `React.ReactChild` | The children that the tooltip belongs to. Must be a single React.child. |
| `className` | `undefined \| string` | Custom className to add to the tooltip overlay container. |
| `content` | `React.ReactNode` | The overlay content for the tooltip. |
| `enterDelay` | `undefined \| number` | Delay in milliseconds before showing the tooltip when interacting via touch or mouse. |
| `leaveDelay` | `undefined \| number` | Delay in milliseconds before hiding the tooltip when interacting via touch or mouse. |
| `open` | `undefined \| false \| true` | Manually control the open state |
| `showArrow` | `undefined \| false \| true` | Whether or not to show an arrow on the Tooltip. Defaults to `false`. |


