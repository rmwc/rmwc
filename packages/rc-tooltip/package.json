{
  "name": "@rmwc/rc-tooltip",
  "version": "14.2.6-alpha.0",
  "description": "RMWC legacy tooltip component",
  "main": "./index.js",
  "types": "./index.d.ts",
  "exports": {
    ".": {
      "import": "./index.mjs",
      "types": "./index.d.ts",
      "require": "./index.js"
    },
    "./styles": {
      "import": "./styles.mjs",
      "types": "./styles.d.ts",
      "require": "./styles.js"
    },
    "./rc-tooltip.css": "./rc-tooltip.css"
  },
  "publishConfig": {
    "access": "public"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/rmwc/rmwc.git"
  },
  "keywords": [
    "rmwc"
  ],
  "author": "rmwc",
  "private": false,
  "license": "MIT",
  "bugs": {
    "url": "https://github.com/rmwc/rmwc/issues"
  },
  "homepage": "https://github.com/rmwc/rmwc/tree/master/packages/rc-tooltip",
  "peerDependencies": {
    "react": ">=16.8.x",
    "react-dom": ">=16.8.x"
  },
  "dependencies": {
    "@rmwc/base": "workspace:*",
    "@rmwc/provider": "workspace:*",
    "rc-tooltip": "6.1.2"
  },
  "devDependencies": {
    "@types/rc-tooltip": "3.7.10"
  }
}
