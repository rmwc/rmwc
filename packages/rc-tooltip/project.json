{
  "name": "rc-tooltip",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "packages/rc-tooltip/src",
  "projectType": "library",
  "tags": [
    "publishable",
    "type:ui-component",
    "readme"
  ],
  "targets": {
    "lint": {
      "executor": "@nx/linter:eslint",
      "outputs": [
        "{options.outputFile}"
      ],
      "options": {
        "lintFilePatterns": [
          "packages/rc-tooltip/**/*.{ts,tsx,js,jsx}"
        ]
      }
    },
    "build": {
      "executor": "@nx/vite:build",
      "dependsOn": [{
        "projects": "readme",
        "target": "generate-markdown"
      }],
      "outputs": [
        "{options.outputPath}"
      ],
      "defaultConfiguration": "production",
      "options": {
        "outputPath": "dist/packages/rc-tooltip"
      },
      "configurations": {
        "development": {
          "mode": "development"
        },
        "production": {
          "mode": "production"
        }
      }
    },
    "test": {
      "executor": "@nx/vite:test",
      "outputs": [
        "{options.reportsDirectory}"
      ],
      "options": {
        "passWithNoTests": true,
        "reportsDirectory": "../../coverage/packages/rc-tooltip"
      }
    },
    "npm-deploy": {
      "executor": "ngx-deploy-npm:deploy",
      "options": {
        "access": "public",
        "buildTarget": "production"
      }
    }
  }
}