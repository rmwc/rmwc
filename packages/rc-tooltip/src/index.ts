export * from './lib/rc-tooltip';
