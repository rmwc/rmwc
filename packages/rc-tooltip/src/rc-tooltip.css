@keyframes rmwcToolTipIn {
  0% {
    opacity: 0;
    transform-origin: 50% 50%;
    transform: scale(0, 0);
  }
  100% {
    opacity: 1;
    transform-origin: 50% 50%;
    transform: scale(1, 1);
  }
}
@keyframes rmwcToolTipOut {
  0% {
    opacity: 1;
  }
  100% {
    opacity: 0;
  }
}

.rmwc-rc-tooltip {
  position: absolute;
  z-index: 9999;
  display: block;
  visibility: visible;
  line-height: 1.5;
  font-size: 0.75rem;
  font-family: inherit;
  padding: 1px;
  opacity: 0.9;
}

.rmwc-rc-tooltip-hidden {
  display: none;
}

.rmwc-rc-tooltip.rmwc-rc-tooltip-zoom-enter,
.rmwc-rc-tooltip.rmwc-rc-tooltip-zoom-leave {
  display: block;
}

.rmwc-rc-tooltip-zoom-enter,
.rmwc-rc-tooltip-zoom-appear {
  opacity: 0;
  animation-duration: 0.15s;
  animation-fill-mode: both;
  animation-play-state: paused;
}

.rmwc-rc-tooltip-zoom-leave {
  animation-duration: 0.075s;
  animation-fill-mode: both;
  animation-play-state: paused;
}

.rmwc-rc-tooltip-zoom-enter.rmwc-rc-tooltip-zoom-enter-active,
.rmwc-rc-tooltip-zoom-appear.rmwc-rc-tooltip-zoom-appear-active {
  animation-name: rmwcToolTipIn;
  animation-play-state: running;
}

.rmwc-rc-tooltip-zoom-leave.rmwc-rc-tooltip-zoom-leave-active {
  animation-name: rmwcToolTipOut;
  animation-play-state: running;
}

.rmwc-rc-tooltip-inner {
  display: flex;
  align-items: center;
  padding: 0.33rem 1rem;
  color: var(--mdc-theme-surface, white);
  text-align: left;
  text-decoration: none;
  background-color: var(--mdc-theme-on-surface, black);
  border-radius: 0.25rem;
  min-height: 2rem;
  box-sizing: border-box;
}

.rmwc-rc-tooltip--show-arrow .rmwc-rc-tooltip-arrow {
  display: block;
}

.rmwc-rc-tooltip-arrow {
  display: none;
}

.rmwc-rc-tooltip-arrow,
.rmwc-rc-tooltip-arrow-inner {
  position: absolute;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
}

.rmwc-rc-tooltip-placement-top .rmwc-rc-tooltip-arrow,
.rmwc-rc-tooltip-placement-topLeft .rmwc-rc-tooltip-arrow,
.rmwc-rc-tooltip-placement-topRight .rmwc-rc-tooltip-arrow {
  bottom: -5px;
  margin-left: -6px;
  border-width: 6px 6px 0;
  border-top-color: var(--mdc-theme-on-surface, black);
}

.rmwc-rc-tooltip-placement-top .rmwc-rc-tooltip-arrow-inner,
.rmwc-rc-tooltip-placement-topLeft .rmwc-rc-tooltip-arrow-inner,
.rmwc-rc-tooltip-placement-topRight .rmwc-rc-tooltip-arrow-inner {
  bottom: 1px;
  margin-left: -6px;
  border-width: 6px 6px 0;
  border-top-color: var(--mdc-theme-surface, white);
}

.rmwc-rc-tooltip-placement-top .rmwc-rc-tooltip-arrow {
  left: 50%;
}

.rmwc-rc-tooltip-placement-topLeft .rmwc-rc-tooltip-arrow {
  left: 15%;
}

.rmwc-rc-tooltip-placement-topRight .rmwc-rc-tooltip-arrow {
  right: 15%;
}

.rmwc-rc-tooltip-placement-right .rmwc-rc-tooltip-arrow,
.rmwc-rc-tooltip-placement-rightTop .rmwc-rc-tooltip-arrow,
.rmwc-rc-tooltip-placement-rightBottom .rmwc-rc-tooltip-arrow {
  left: -5px;
  margin-top: -6px;
  border-width: 6px 6px 6px 0;
  border-right-color: var(--mdc-theme-on-surface, black);
}

.rmwc-rc-tooltip-placement-right .rmwc-rc-tooltip-arrow-inner,
.rmwc-rc-tooltip-placement-rightTop .rmwc-rc-tooltip-arrow-inner,
.rmwc-rc-tooltip-placement-rightBottom .rmwc-rc-tooltip-arrow-inner {
  left: 1px;
  margin-top: -6px;
  border-width: 6px 6px 6px 0;
  border-right-color: var(--mdc-theme-surface, white);
}

.rmwc-rc-tooltip-placement-right .rmwc-rc-tooltip-arrow {
  top: 50%;
}

.rmwc-rc-tooltip-placement-rightTop .rmwc-rc-tooltip-arrow {
  top: 15%;
  margin-top: 0;
}

.rmwc-rc-tooltip-placement-rightBottom .rmwc-rc-tooltip-arrow {
  bottom: 15%;
}

.rmwc-rc-tooltip-placement-left .rmwc-rc-tooltip-arrow,
.rmwc-rc-tooltip-placement-leftTop .rmwc-rc-tooltip-arrow,
.rmwc-rc-tooltip-placement-leftBottom .rmwc-rc-tooltip-arrow {
  right: -5px;
  margin-top: -6px;
  border-width: 6px 0 6px 6px;
  border-left-color: var(--mdc-theme-on-surface, black);
}

.rmwc-rc-tooltip-placement-left .rmwc-rc-tooltip-arrow-inner,
.rmwc-rc-tooltip-placement-leftTop .rmwc-rc-tooltip-arrow-inner,
.rmwc-rc-tooltip-placement-leftBottom .rmwc-rc-tooltip-arrow-inner {
  right: 1px;
  margin-top: -6px;
  border-width: 6px 0 6px 6px;
  border-left-color: var(--mdc-theme-surface, white);
}

.rmwc-rc-tooltip-placement-left .rmwc-rc-tooltip-arrow {
  top: 50%;
}

.rmwc-rc-tooltip-placement-leftTop .rmwc-rc-tooltip-arrow {
  top: 15%;
  margin-top: 0;
}

.rmwc-rc-tooltip-placement-leftBottom .rmwc-rc-tooltip-arrow {
  bottom: 15%;
}

.rmwc-rc-tooltip-placement-bottom .rmwc-rc-tooltip-arrow,
.rmwc-rc-tooltip-placement-bottomLeft .rmwc-rc-tooltip-arrow,
.rmwc-rc-tooltip-placement-bottomRight .rmwc-rc-tooltip-arrow {
  top: -5px;
  margin-left: -6px;
  border-width: 0 6px 6px;
  border-bottom-color: var(--mdc-theme-on-surface, black);
}

.rmwc-rc-tooltip-placement-bottom .rmwc-rc-tooltip-arrow-inner,
.rmwc-rc-tooltip-placement-bottomLeft .rmwc-rc-tooltip-arrow-inner,
.rmwc-rc-tooltip-placement-bottomRight .rmwc-rc-tooltip-arrow-inner {
  top: 1px;
  margin-left: -6px;
  border-width: 0 6px 6px;
  border-bottom-color: var(--mdc-theme-surface, white);
}

.rmwc-rc-tooltip-placement-bottom .rmwc-rc-tooltip-arrow {
  left: 50%;
}

.rmwc-rc-tooltip-placement-bottomLeft .rmwc-rc-tooltip-arrow {
  left: 15%;
}

.rmwc-rc-tooltip-placement-bottomRight .rmwc-rc-tooltip-arrow {
  right: 15%;
}

@media (min-width: 768px) {
  .rmwc-rc-tooltip-inner {
    padding: 0.33rem 0.5rem;
    min-height: 1.5rem;
  }
}
