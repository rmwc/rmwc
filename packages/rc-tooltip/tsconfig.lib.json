{
  "extends": "./tsconfig.json",
  "compilerOptions": {
    "outDir": "../../dist/out-tsc",
    "types": ["node", "vite/client"]
  },
  "exclude": [
    "**/*.spec.ts",
    "**/*.test.ts",
    "**/*.spec.tsx",
    "**/*.test.tsx",
    "**/*.spec.js",
    "**/*.test.js",
    "**/*.spec.jsx",
    "**/*.test.jsx",
    "**/*.story.tsx"
  ],
  "include": ["src/**/*.js", "src/**/*.jsx", "src/**/*.ts", "src/**/*.tsx"]
}
