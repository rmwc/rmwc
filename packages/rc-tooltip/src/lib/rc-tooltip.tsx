import React from 'react';
import { classNames } from '@rmwc/base';
import ReactTooltip from 'rc-tooltip';
import { useProviderContext } from '@rmwc/provider';

import "../rc-tooltip.css";

export type RCTooltipActivationT = 'hover' | 'click' | 'focus';

export type RCTooltipAlignT =
  | 'left'
  | 'right'
  | 'top'
  | 'bottom'
  | 'topLeft'
  | 'topRight'
  | 'bottomLeft'
  | 'bottomRight';

/** A Tooltip component for displaying informative popover information. */
export interface RCTooltipProps {
  /** The overlay content for the tooltip. */
  content: React.ReactNode;
  /** The children that the tooltip belongs to. Must be a ReactElement. */
  children: React.ReactElement;
  /** Activate the tooltip through one or more interactions. Defaults to `['hover', 'focus']`. */
  activateOn?: RCTooltipActivationT | RCTooltipActivationT[];
  /** Whether or not to show an arrow on the Tooltip. Defaults to `false`. */
  showArrow?: boolean;
  /** Custom className to add to the tooltip overlay container. */
  className?: string;
  /** Delay in milliseconds before showing the tooltip when interacting via touch or mouse. */
  enterDelay?: number;
  /** Delay in milliseconds before hiding the tooltip when interacting via touch or mouse. */
  leaveDelay?: number;
  /** How to align the tooltip. Defaults to `top`. */
  align?: RCTooltipAlignT;
  /** Manually control the open state */
  open?: boolean;
}

/** A Tooltip component for displaying informative popover information. */
export const RCTooltip = function RCTooltip({
  children,
  content,
  className,
  open,
  ...rest
}: RCTooltipProps) {
  const providerContext = useProviderContext();

  // merge together provider options
  const {
    showArrow = false,
    enterDelay = 0,
    leaveDelay = 0,
    align = 'top',
    activateOn = ['hover', 'focus'] as RCTooltipActivationT[]
  } = {
    ...providerContext.tooltip,
    ...rest
  };

  return (
    <ReactTooltip
      {...(open !== undefined ? { visible: open } : {})}
      trigger={Array.isArray(activateOn) ? activateOn : [activateOn]}
      prefixCls="rmwc-rc-tooltip"
      placement={align}
      transitionName="rmwc-rc-tooltip-zoom"
      mouseEnterDelay={enterDelay / 1000}
      mouseLeaveDelay={leaveDelay / 1000}
      overlay={content}
      overlayClassName={classNames(className, {
        'rmwc-rc-tooltip--show-arrow': showArrow
      })}
      destroyTooltipOnHide
    >
      {children}
    </ReactTooltip>
  );
};
