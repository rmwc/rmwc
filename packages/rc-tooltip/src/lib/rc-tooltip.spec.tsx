import { RMWCProvider } from '@rmwc/provider';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { RCTooltip } from './rc-tooltip';

describe('Tooltip', () => {
  it('renders', async () => {
    const { asFragment } = render(
      <RCTooltip content="tooltip">
        <span>test</span>
      </RCTooltip>
    );
    await userEvent.hover(screen.getByText('test'));
    expect(screen.getByText('tooltip')).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it('activateOn', () => {
    render(
      <RCTooltip content="tooltip" activateOn="click">
        <span>test</span>
      </RCTooltip>
    );
  });

  it('showArrow', () => {
    const { asFragment } = render(
      <RCTooltip content="tooltip" showArrow>
        <span>test</span>
      </RCTooltip>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('className', () => {
    render(
      <RCTooltip content="tooltip" className="my-custom-classname">
        <span>test</span>
      </RCTooltip>
    );
  });

  it('enterDelay', () => {
    render(
      <RCTooltip content="tooltip" enterDelay={1000}>
        <span>test</span>
      </RCTooltip>
    );
  });

  it('leaveDelay', () => {
    render(
      <RCTooltip content="tooltip" leaveDelay={1000}>
        <span>test</span>
      </RCTooltip>
    );
  });

  it('align', () => {
    const { asFragment } = render(
      <RCTooltip content="tooltip" align="bottom">
        <span>test</span>
      </RCTooltip>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('works with provider', () => {
    render(
      <RMWCProvider tooltip={{}}>
        <RCTooltip content="tooltip">
          <span>test</span>
        </RCTooltip>
      </RMWCProvider>
    );
  });

  it('accepts allowed align value from provider context', () => {
    vi.spyOn(console, 'warn');
    render(
      <RMWCProvider tooltip={{ align: 'bottom' }}>
        <RCTooltip content="tooltip">
          <span>test</span>
        </RCTooltip>
      </RMWCProvider>
    );
    expect(console.warn).not.toBeCalled();
  });

  it('warn against wrong align value from provider context', () => {
    vi.spyOn(console, 'warn');
    render(
      <RMWCProvider tooltip={{ align: 'start' }}>
        <RCTooltip content="tooltip">
          <span>test</span>
        </RCTooltip>
      </RMWCProvider>
    );
    expect(console.warn).toBeCalled();
  });
});
