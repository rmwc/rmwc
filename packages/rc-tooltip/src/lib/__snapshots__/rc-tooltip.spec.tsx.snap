// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tooltip > align 1`] = `
<DocumentFragment>
  <span>
    test
  </span>
</DocumentFragment>
`;

exports[`Tooltip > renders 1`] = `
<DocumentFragment>
  <span>
    test
  </span>
</DocumentFragment>
`;

exports[`Tooltip > showArrow 1`] = `
<DocumentFragment>
  <span>
    test
  </span>
</DocumentFragment>
`;
